src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs

```diff
@@ -19,44 +20,39 @@ namespace Microsoft.CodeAnalysis.QuickInfo
 {
     internal abstract partial class CommonSemanticQuickInfoProvider : CommonQuickInfoProvider
     {
-        protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(
-            Document document,
-            SyntaxToken token,
-            CancellationToken cancellationToken)
+        protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(QuickInfoContext context)
         {
-            var (model, tokenInformation, supportedPlatforms) = await ComputeQuickInfoDataAsync(document, token, cancellationToken).ConfigureAwait(false);
+            var (model, tokenInformation, supportedPlatforms) = await ComputeQuickInfoDataAsync(context).ConfigureAwait(false);
 
             if (tokenInformation.Symbols.IsDefaultOrEmpty)
             {
                 return null;
             }
 
-            return await CreateContentAsync(document.Project.Solution.Workspace,
-                token, model, tokenInformation, supportedPlatforms,
-                cancellationToken).ConfigureAwait(false);
+            return await CreateContentAsync(context.Workspace,
```

Comment:
😕 It's not clear how we can rely on the presence of a `Workspace` but cannot rely on the presence of a `Document`

Reply:
This is the pattern that we followed even for other features used in LSIF generator. Workspace is needed to access language services, workspace services and such, which is available to LSIF generator as it directly references Workspaces/Features layer. However, there is no Document/Project, as it works off a compilation (though we can simulate it).